wandb/sdk/wandb_artifacts.py

```diff
@@ -287,37 +272,25 @@ def finalize(self):
         if self._final:
             return self._file_entries
 
-        # Record any created files in the manifest.
-        if self._added_new:
-            self.add_dir(self._artifact_dir.name)
-
         # mark final after all files are added
         self._final = True
         self._digest = self._manifest.digest()
 
-        # If there are new files, move them into the artifact cache now. Our temp
-        # self._artifact_dir may not be available by the time file pusher syncs
-        # these files.
-        if self._added_new:
-            # Update the file entries for new files to point at their new location.
-            def remap_entry(entry):
-                if entry.local_path is None or not entry.local_path.startswith(
-                    self._artifact_dir.name
-                ):
-                    return entry
-                rel_path = os.path.relpath(
-                    entry.local_path, start=self._artifact_dir.name
-                )
-                local_path = os.path.join(self._artifact_dir.name, rel_path)
-                cache_path, hit = self._cache.check_md5_obj_path(
-                    entry.digest, entry.size
-                )
-                if not hit:
-                    shutil.copyfile(local_path, cache_path)
-                entry.local_path = cache_path
+    def _write_through(self, name, path, digest=None):
```

Comment:
nit: the name `_write_through` may be improved by naming something like `_write_local_file`, or `_add_local_file` ... something like this.

Reply:
Good call, just renamed.